src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -286,7 +287,43 @@ app.directive('dashboardApp', function ($injector) {
         }).catch(notify.fatal);
       };
 
+      $scope.showFilterBar = () => filterBar.getFilters().length > 0 || !$scope.fullScreenMode;
+      let onRouteChange;
+      const setFullScreenMode = (fullScreenMode) => {
+        $scope.fullScreenMode = fullScreenMode;
+        dashboardState.setFullScreenMode(fullScreenMode);
+        chrome.setVisible(!fullScreenMode);
+        $scope.$root.$broadcast('globalNav:update');
+        if (fullScreenMode) {
+          onRouteChange = $scope.$on('$routeChangeStart', () => chrome.setVisible(true));
```

Comment:
Gotcha. When the user navigates away, how is the `onRouteChange`/deregistration called?